### PR TITLE
[FT] Custom op supports eager mode

### PR DESCRIPTION
**Pull request by FrostML**
### PR types

Breaking changes 

### PR changes

APIs

### Description

Custom op supports eager mode for PaddlePaddle develop.

### CODE REVIEW

File: paddlenlp/ops/faster_transformer/transformer/decoder.py [hunk 1]
```diff
@@ -24,6 +24,10 @@
 from paddlenlp.ops.ext_utils import load, LOADED_EXT
 from paddlenlp.ops import transfer_param
 
+if paddle.__version__ == "0.0.0":
+    from paddle.framework import (core, in_dygraph_mode, _non_static_mode,
+                                  _dygraph_tracer, _in_legacy_dygraph)
+
```
Comment (guoshengCS) [inline paddlenlp/ops/faster_transformer/transformer/decoder.py:31]: 最好不要用"0.0.0"检测，不见得是"0.0.0"就可以

File: paddlenlp/ops/faster_transformer/transformer/decoding.py [hunk 2]
```diff
@@ -42,75 +46,199 @@ def infer_transformer_decoding(
...
+        outputs = {
+            'OutputIds': output_ids,
+            'ParentIds': parent_ids,
+            'SequenceLength': sequence_length
+        }
+
+        helper.append_op(type="fusion_decoding",
+                         inputs=inputs,
+                         outputs=outputs,
+                         attrs=attrs)
```
Comment (guoshengCS) [inline paddlenlp/ops/faster_transformer/transformer/decoding.py]: 这种还是看看抽象组织成能复用的函数吧，避免每个地方都去写这么一大段重复代码

File: paddlenlp/ops/faster_transformer/transformer/decoder.py (hunk 1 above)
Comment (FrostML) [reply paddlenlp/ops/faster_transformer/transformer/decoder.py:31]: Done. Thanks.

File: paddlenlp/ops/faster_transformer/transformer/decoding.py (hunk 2 above)
Comment (FrostML) [reply paddlenlp/ops/faster_transformer/transformer/decoding.py]: Done. Thanks.

File: paddlenlp/ops/faster_transformer/transformer/encoder.py [hunk 3]
```diff
@@ -344,14 +368,15 @@ def init_func(layer):
...
+    encoder_lib = "/paddle/fast_transformer/helper/PaddleNLP/paddlenlp/ops/build/lib/libdecoding_op.so"
     if not self.training:
         try:
             # Pass decoding lib to prevent re-building encoder.
             # Todo: check weather decoding lib have contained encoder or not.
             if encoder_lib is not None:
                 if "FasterTransformer" not in LOADED_EXT.keys():
                     ops = paddle.utils.cpp_extension.load_op_meta_info_and_register_op(
-                        decoding_lib)
+                        encoder_lib)
```
Comment (guoshengCS) [inline paddlenlp/ops/faster_transformer/transformer/encoder.py:378]: 上面 `encoder_lib = "/paddle/fast_transformer/helper/PaddleNLP/paddlenlp/ops/build/lib/libdecoding_op.so"` 这个是不是忘删掉了

Comment (FrostML) [reply paddlenlp/ops/faster_transformer/transformer/encoder.py:378]: Done. Thanks.

**Review by guoshengCS — APPROVED** [review]
LGTM

**Pull request merged**